.github/monitor-typescript-errors/index.js

```diff
@@ -0,0 +1,61 @@
+const { getOctokit, context } = require( '@actions/github' );
+const { setFailed, getInput } = require( '@actions/core' );
+const { openFile } = require( './utils/file' );
+const { parseXml, getFilesWithNewErrors } = require( './utils/xml' );
+const { generateMarkdownMessage } = require( './utils/markdown' );
+const { addRecord } = require( './utils/airtable' );
+
+const runner = async () => {
+	const token = getInput( 'repo-token', { required: true } );
+	const octokit = getOctokit( token );
+	const payload = context.payload;
+	const repo = payload.repository.name;
+	const owner = payload.repository.owner.login;
+	const fileName = getInput( 'compare', {
+		required: true,
+	} );
+	const newCheckStyleFile = openFile( fileName );
+	const newCheckStyleFileParsed = parseXml( newCheckStyleFile );
+	const currentCheckStyleFilePath =
+		getInput( 'trunk_branch_folder', { required: true } ) + '/' + fileName;
+
+	if ( ! currentCheckStyleFilePath ) {
+		return;
+	}
+
+	const currentCheckStyleFile = openFile( currentCheckStyleFilePath );
+	const currentCheckStyleFileParsed = parseXml( currentCheckStyleFile );
+	const { header } = generateMarkdownMessage( newCheckStyleFileParsed );
+	const filesWithNewErrors = getFilesWithNewErrors(
+		newCheckStyleFileParsed,
+		currentCheckStyleFileParsed
+	);
+
+	const message =
+		header +
+		'\n' +
+		( filesWithNewErrors.length > 0
+			? `⚠️ ⚠️ This PR introduces new TS errors on ${ filesWithNewErrors.length } files: \n` +
+			  '<details> \n' +
+			  filesWithNewErrors.join( '\n' ) +
```

Comment:
we'd need to join it with `<br>` maybe? new line char doesn't seem to put file paths on separate lines. 

Also it should be possible to actually display new errors on the files like eslint errors show up now at the end of the PR files review I believe but it could be next steps :)